macros/generic_tests/sequential_values.sql

```diff
@@ -1,18 +1,25 @@
-{% test sequential_values(model, column_name, interval=1, datepart=None) %}
+{% test sequential_values(model, column_name, interval=1, datepart=None, group_by_columns = []) %}
 
-  {{ return(adapter.dispatch('test_sequential_values', 'dbt_utils')(model, column_name, interval, datepart)) }}
+  {{ return(adapter.dispatch('test_sequential_values', 'dbt_utils')(model, column_name, interval, datepart, group_by_columns)) }}
 
 {% endtest %}
 
-{% macro default__test_sequential_values(model, column_name, interval=1, datepart=None) %}
+{% macro default__test_sequential_values(model, column_name, interval=1, datepart=None, group_by_columns = []) %}
 
 {% set previous_column_name = "previous_" ~ dbt_utils.slugify(column_name) %}
 
+{% if group_by_columns|length() > 0 %}
+  {% set select_gb_cols = group_by_columns|join(' ,') + ', ' %}
+  {% set partition_gb_cols = 'partition by ' + group_by_columns|join(',') %}
```

Comment:
_this_ one's a real nitpick - why `' ,'` instead of `', '`? This will render as select `col_1 ,col_2` right? Obviously doesn't break anything.

Reply:
Excellent question. Indefensible on my part! 🙂 Fixed.